hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/pom.xml

```diff
@@ -27,6 +27,8 @@
 
     <properties>
         <grpc.version>1.69.0</grpc.version>
+        <!-- Last version that works with Centos 7 -->
+        <grpc.plugin.version>1.65.0</grpc.plugin.version>
```

Comment:
It looks risky to use a different gRPC version for codegen and runtime, an alternative I can imagine -track the generated source code in Git, and regenerate the code on each gRPC upgrading

Reply:
> There are no code changes in the compiler between those versions

it's fine for these specific version pair, but may not be good for future upgrading

> committing generated code to git is troublesome, I would very much like to avoid that.

I tend to agree with you in most cases, but I think for this case, it's a most safe and relatively cheap way.

Reply:
> it's fine for these specific version pair, but may not be good for future upgrading
> 

We can cross that rover when we get there.

If you REALLY don't want to use different versions, then I'd preder recompiling the plugin from source on Centos 7, though I'd hate to put in that much effort to support EOL distros.

Reply:
Another way used by Spark, allow user to configure pre-installed `protoc`/`grpc` commands to generate the code, then we can compile and install those commands from the source code on the CentOS 7 build image (I'm not familiar with the native world, so have no idea how much maintenance effort will be paid for this way)

Reply:
Yes, that's what I meant by recompiling the grpc compiler from source.

Reply:
BTW, Hadoop actually has committed a few generated code into Git repo
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/arm-java

Reply:
Thanks to both of you for the discussion! In the current situation, I lean towards supporting Story's approach, as long as it doesn't block the progress of Hadoop 3.4.2. In fact, we don't actually need this CentOS7 script during the release process. If a similar issue arises in a future version of Ubuntu, I believe we can follow @pan3793  method to fix it.